src/api/api-util.ts

```diff
@@ -0,0 +1,46 @@
+import {HttpErrorResponse} from "@angular/common/http";
+
+// Example HTTP 400 Zod error
+/*
+[
+  {
+    "type": "Body",
+    "errors": {
+      "issues": [
+        {
+          "validation": "email",
+          "code": "invalid_string",
+          "message": "Invalid email",
+          "path": [
+            "email"
+          ]
+        }
+      ],
+      "name": "ZodError"
+    }
+  }
+]
+*/
+
+/**
+ * Get user-friendly error message from API error.
+ * @param error
+ */
```

Comment:
Okay - just noting this is just for the 400 error response from the zod middleware on express. 

The general error class is here

```
const errorResponse: ErrorResponse = {
    status: statusCode,
    message: message,
  };
```

I would prefer to have the validation error from zod report a 422 (parse error) - this is a bit easier to isolate. Anyway thanks for pulling this out.